src/sveltekit/index.ts

```diff
@@ -0,0 +1,57 @@
+import { pathToFileURL } from 'url'
+import { installPolyfills } from '@sveltejs/kit/node/polyfills'
+import type { Request } from 'firebase-functions/v2/https'
+import type { Response } from 'express'
+import type { IncomingHttpHeaders } from 'http'
+
+const kitPromise = import(`${pathToFileURL(process.cwd())}/index.js`)
+const manifestPromise = import(`${pathToFileURL(process.cwd())}/manifest.js`)
+
+// Request, Response, fetch, etc.
+installPolyfills()
+
+export const handle = async (req: Request, res: Response) => {
+  const [{ Server }, { manifest }] = await Promise.all([
+    kitPromise,
+    manifestPromise
+  ])
+
+  const server = new Server(manifest)
+  await server.init({ env: process.env })
+  const rendered = await server.respond(toSvelteKitRequest(req))
+
+  if (!rendered) {
+    return res.writeHead(404, 'Not Found').end()
+  }
+
+  const body = await rendered.text()
+  return res
+    .writeHead(rendered.status, Object.fromEntries(rendered.headers))
+    .end(body)
+}
+
+// https://github.com/jthegedus/svelte-adapter-firebase/blob/main/src/files/firebase-to-svelte-kit.js
+function toSvelteKitRequest(request: Request) {
+  // Firebase sometimes omits the protocol used. Default to http.
```

Comment:
gen2 functions should handle x-forwarded correctly, so these links shouldn't be needed any longer

Reply:
From what I can tell, this seems to be on the only way to get the actual full `host` to pass as the base in the `URL` constructor that follows. I couldn't find a way to piece  together the `host` without using these headers.